core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -790,34 +823,33 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     }
 
     while (cur ne null) {
-      if (RuntimeFlags.opSupervision(runtimeFlags)) {
-        self.getSupervisor().onEffect(self, cur)
-      }
+
+      cur = drainQueueWhileRunning(runtimeFlags, lastTrace, cur)
 
       val nextTrace = cur.trace
       if (nextTrace ne Trace.empty) lastTrace = nextTrace
 
-      cur = drainQueueWhileRunning(runtimeFlags, lastTrace, cur)
+      if (RuntimeFlags.opSupervision(runtimeFlags)) {
+
+        self.getSupervisor().onEffect(self, cur)
```

Comment:
I am not sure this is related to op-logging and I think it only belongs in one place: when pattern matching on the current instruction in the run loop.